keyvi/include/keyvi/dictionary/dictionary_compiler.h

```diff
@@ -162,6 +162,11 @@ class DictionaryCompiler final {
     }
 
     std::ofstream out_stream(filename, std::ios::binary);
+
+    if (!out_stream.good()) {
+      throw std::invalid_argument("failed to open file");
```

Comment:
Maybe we can include file-name in the message as well ? 
Also seems we are duplication the message in 4 places, prob we can have some either method or generic message to avoid duplication ?